telegram/ext/dispatcher.py

```diff
@@ -229,30 +236,89 @@ def _pooled(self):
                 break
 
             promise.run()
+
+            if not promise.exception:
+                self.update_persistence(update=promise.update)
+                continue
+
             if isinstance(promise.exception, DispatcherHandlerStop):
                 self.logger.warning(
                     'DispatcherHandlerStop is not supported with async functions; func: %s',
                     promise.pooled_function.__name__)
+                continue
+
+            if promise.pooled_function in self.error_handlers:
+                self.logger.error('An error was raised while processing the update '
+                                  'and an uncaught error was raised while handling '
+                                  'the error with an error_handler')
+                continue
+
+            if promise.error_handler:
+                try:
+                    context = CallbackContext.from_error(promise.update, promise.exception,
+                                                         self)
+                    if promise.update:
+                        promise.error_handler(promise.update, context, *promise.args,
+                                              **promise.kwargs)
+                    else:
+                        promise.error_handler(context, *promise.args, **promise.kwargs)
+                except Exception:
+                    self.logger.exception('An error was raised while processing the update '
+                                          'and an uncaught error was raised while handling '
+                                          'the error with an error_handler')
+            elif promise.update:
+                try:
+                    self.dispatch_error(promise.update, promise.exception)
+                except Exception:
+                    self.logger.exception('An error was raised while processing the update '
+                                          'and an uncaught error was raised while handling '
+                                          'the error with an error_handler')
+            else:
+                self.logger.exception(
+                    'No error handlers are set for async function, logging exception.',
+                    exc_info=promise.exception)
```

Comment:
1. Use `continue`.
2. Shorter log message.

```suggestion
                continue
                
            self.logger.exception('Caught unhandled exception from an async function',
                                  exc_info=promise.exception)
```